authz.go

```diff
@@ -225,6 +237,23 @@ func (a *authorizer) authorize(ctx context.Context, sr authz.SubjectAccessReview
 			break
 		}
 	}
+
+	// if cache and dry run are enabled, error need to be added to log if authorize result from cache and authorize result from zts / zms are different.
+	if a.AuthorizationConfig.Config.UseCache {
+		if decision.status.Allowed != granted {
```

Comment:
if `status` is false from line 202, would `decision` be nil?

Reply:
decision will be nil, and in this case we should not check for mismatch, let me update the if statement.